cpp/src/pca/pca.cuh

```diff
@@ -67,6 +68,20 @@ void truncCompExpVars(const raft::handle_t& handle,
                                 prms.n_components,
                                 std::size_t(1),
                                 stream);
+
+  // Compute the scalar noise_vars defined as (pseudocode)
+  // (n_components < min(n_cols, n_rows)) ? explained_var_all[n_components:].mean() : 0
+  if (prms.n_components < prms.n_cols && prms.n_components < prms.n_rows) {
+    raft::stats::mean(noise_vars,
+                      explained_var_all.data() + prms.n_components,
+                      std::size_t{1},
+                      prms.n_cols - prms.n_components,
+                      false,
+                      true,
+                      stream);
+  } else {
+    raft::matrix::setValue(noise_vars, noise_vars, math_t{0}, 1, stream);
```

Comment:
`noise_vars` is a len-1 device array that I want to set to 0. This seems to work (and is used at least one other place in cuml already), but I'm honestly not sure if it's the best spelling of that. In particular, I don't understand why a method for filling an output array with a single value also takes in an input array (which is the same as the output array in all calling locations I can find).